MekHQ/src/mekhq/campaign/rating/CampaignOpsReputation.java

```diff
@@ -503,22 +503,41 @@ public int getTransportValue() {
         TransportCapacityIndicators tci = new TransportCapacityIndicators();
         tci.updateCapacityIndicators(getMechBayCount(), getMechCount());
         tci.updateCapacityIndicators(getProtoBayCount(), getProtoCount());
-        tci.updateCapacityIndicators(getHeavyVeeBayCount(), getHeavyVeeCount());
-
-        // vehicles are the only units that can share heavy/light bays so we have some special logic
-        // we've stuffed all possible heavy vehicles into heavy vehicle bays.
-        // if we have some heavy vehicle bays left over, add them to the light vehicle bay count
-        int heavyVeeBays = getHeavyVeeBayCount() - getHeavyVeeCount();
-        int lightVeeBays = getLightVeeBayCount();
-        if(heavyVeeBays > 0) {
-            lightVeeBays += heavyVeeBays;
-        }
-
-        tci.updateCapacityIndicators(lightVeeBays, getLightVeeCount());
-        tci.updateCapacityIndicators(getFighterBayCount(), getFighterCount());
         tci.updateCapacityIndicators(getBaBayCount(), getBattleArmorCount() / 5); // battle armor bays can hold 5 battle armor per bay
         tci.updateCapacityIndicators(getInfantryBayCount(), calcInfantryPlatoons());
-        tci.updateCapacityIndicators(getSmallCraftBayCount(), getSmallCraftCount());
+
+        // Light vehicles can use heavy or light vehicle bays, while fighters can use fighter or small craft bays
+        // Once we've stuffed stuffed all possible heavy vehicles into heavy vehicle bays.
+        // if we have some heavy vehicle bays left over, add them to the light vehicle bay count, and then calculate the
+        // number of heavy vehicle bays that are still empty.
+        // The same is done for small craft and fighters, just replace heavy vehicle with small craft and light vehicle with fighters
+        int excessHeavyVeeBays = Math.max(getHeavyVeeBayCount() - getHeavyVeeCount(), 0);
+        int excessSmallCraftBays = Math.max(getSmallCraftBayCount() - getSmallCraftCount(), 0);
+
+        // We need to subtract any filled bays from the count. This follows the following logic:
+        // Assume you have 2 heavy vehicle bays, and 4 light vehicle bays, and are trying to store 1 heavy and 5 light vehicles
+        // You have 1 more light vehicle than light vehicle bays to store them in, so you check how many free heavy vehicle bays
+        // there are. Finding 1, you can store the light vehicle there, and it doesn't count as having excess
```

Comment:
@NickAragua 
Is this a sufficient explanation?

Reply:
Sounds good.